userspace/libsinsp/sinsp.h

```diff
@@ -995,7 +1011,7 @@ VISIBILITY_PRIVATE
 	uint32_t m_nevts;
 	int64_t m_filesize;
 
-	scap_mode_t m_mode;
+	scap_mode_t m_mode = SCAP_MODE_LIVE;
```

Comment:
Is this safe?

You might want to add a new `SCAP_MODE_XXX` for unknown or otherwise unset.

Reply:
I changed the way I accomplished this. In production code, this is set by open so I made a way for for the override of open to just explicitly set the mode.